src/Symfony/Component/Serializer/Normalizer/FlattenExceptionNormalizer.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use Symfony\Component\ErrorHandler\Exception\FlattenException;
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
+
+/**
+ * Normalizes FlattenException instances.
+ *
+ * @author Pascal Luna <skalpa@zetareticuli.org>
+ *
+ * @experimental
+ */
+class FlattenExceptionNormalizer implements NormalizerInterface, DenormalizerInterface, CacheableSupportsMethodInterface
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @throws InvalidArgumentException
+     */
+    public function normalize($object, $format = null, array $context = [])
+    {
+        if (!$object instanceof FlattenException) {
+            throw new InvalidArgumentException(sprintf('The object must be an instance of "%s".', FlattenException::class));
+        }
+        /* @var FlattenException $object */
+
+        $normalized = [
+            'detail' => $object->getMessage(),
+            'code' => $object->getCode(),
+            'headers' => $object->getHeaders(),
+            'class' => $object->getClass(),
+            'file' => $object->getFile(),
+            'line' => $object->getLine(),
+            'previous' => null === $object->getPrevious() ? null : $this->normalize($object->getPrevious(), $format, $context),
```

Comment:
shouldnt this actually use the chained normalizer? Thus make it normalizer aware

Reply:
Can you give me an example here?

Reply:
see `NormalizerAwareTrait + Interface` :) (i believe injecting leads to all sort of recursion issues).

So then you can leverage `$this->normalizer->normalize(..)` instead.

Reply:
Not sure it's worth it as we know that previous is a `FlattenException` exception anyway.

Reply:
yes, but if you write a custom decorating normalizer you want the inheritance to be taken care of i figured.

Reply:
Fair enough.

Reply:
Already added the trait.

Reply:
But here, you are using `$this`, you should use `$this->normalizer`, right?